### PR TITLE
Add timeout option to SQL CLI tool.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
Add response timeout option to SQL CLI tool. This is very handy for SQL plugin debugging purposes.
Imagine you are debugging a query and placed a breakpoint into the plugin code. While you are going step-by-step debugging, CLI times out.

See team review discussion in https://github.com/Bit-Quill/opensearch-project-sql/pull/153.

New command line optional argument added:
```
-t <value in seconds>
```
If not given, SQL CLI works as usual. Default timeout is 10 sec.

**Before**
```
$ opensearchsql
...
opensearchsql> select 1;
fetched rows / total rows = 1/1      <--- response in 0-10 sec
+-----+
| 1   |
|-----|
| 1   |
+-----+
opensearchsql> select 1;
Reconnecting...
Reconnected! Please run query again      <--- response in 10-19 sec
opensearchsql> select 1;
Reconnecting...
Connection Failed. Check your OpenSearch is running and then come back     <---- 20+ sec
ConnectionTimeout('TIMEOUT', "HTTPConnectionPool(host='localhost', port=9200): Read timed out. (read timeout=10)", ReadTimeoutError("HTTPConnectionPool(host='localhost', port=9200): Read timed out. (read timeout=10)"))
opensearchsql>
```
**After**
```
$ opensearchsql -t 10000
...
opensearchsql> select 1;
fetched rows / total rows = 1/1          <---- 30 min of intensive debugging with breaks - no timeout!
+-----+
| 1   |
|-----|
| 1   |
+-----+
```

Parameter `request_timeout` is processed by underlying library:
https://github.com/opensearch-project/opensearch-py/blob/8879644ec404b3ab15b739b97775d7bd7e51018b/opensearchpy/transport.py#L457

### How to test
Ref: https://github.com/opensearch-project/sql/blob/2.x/sql-cli/development_guide.md#development-environment-set-up

Do once:
```
cd sql-cli
pip install virtualenv
virtualenv venv
pip install --editable .
```
Then every time you want to debug SQL CLI:
```
<checkout>
cd sql-cli
source ./venv/bin/activate
```
To exit from `venv`:
```
deactivate
```
You can run SQL CLI and switch branch. It will keep working with given timeout until closed.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql-cli/src/opensearch_sql_cli/main.py [hunk 1]
```diff
@@ -71,6 +71,14 @@
     default="sql",
     help="SQL OR PPL",
 )
+@click.option(
+    "-t",
+    "--timeout",
+    "response_timeout",
+    type=click.INT,
+    default=10,
+    help="Timeout to await a response from the server"
```
Comment (joshuali925) [inline sql-cli/src/opensearch_sql_cli/main.py:80]: should the help message indicate it's in seconds? and is the default 10 seconds too short? I think frontend request times out 30 seconds by default

Comment (Yury-Fridlyand) [reply sql-cli/src/opensearch_sql_cli/main.py:80]: 10 sec is default timeout in PY client.

Comment (Yury-Fridlyand) [reply sql-cli/src/opensearch_sql_cli/main.py]: Help message updated in dad0f8f86.

**Review by joshuali925 — APPROVED** [review]

**Review by derek-ho — APPROVED** [review]
LGTM

**Review by penghuo — APPROVED** [review]
Thanks.

**Pull request merged**